boa/test/strategies.py

```diff
@@ -98,13 +101,22 @@ def generate_random_string(n):
 
 
 @_exclude_filter
-def _address_strategy(length: Optional[int] = 100) -> SearchStrategy:
+def _address_strategy(length: Optional[int] = 100, environment: Optional[EnvType] = None) -> SearchStrategy:
+
+    if environment is None:
+        environment = boa.env
+
+    # TODO: add addresses generated by the env through the env.generate_address() method
+    env_accounts = (
+            {obj.address for obj in environment._code_registry.values()}
+    )
+
+    # generate `length` random addresses
     random_strings = generate_random_string(length)
-    # TODO: add addresses from the environment. probably everything in
-    # boa.env._contracts, boa.env._blueprints and boa.env.eoa.
-    accounts = [format_addr(i) for i in random_strings]
+    random_accounts = {format_addr(i) for i in random_strings}
+
     return _DeferredStrategyRepr(
-        lambda: st.sampled_from(list(accounts)[:length]), "accounts"
+        lambda: st.sampled_from(list(env_accounts | random_accounts)), "accounts"
```

Comment:
i think using `st.one_of(st.sampled_from(list(env_accounts), list(random_accounts)))` will generate a slightly better distribution

we can also use `st.binary()` for the random addresses actually